shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java

```diff
@@ -241,27 +241,28 @@ public void setPaddingTopToZeroForFullscreenMode() {
     assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingTop);
 
     // Then we simulate the system applying a window inset.
-    WindowInsets windowInsets = mock(WindowInsets.class);
-    when(windowInsets.getSystemWindowInsetTop()).thenReturn(100);
-    when(windowInsets.getSystemWindowInsetBottom()).thenReturn(100);
-    when(windowInsets.getSystemWindowInsetLeft()).thenReturn(100);
-    when(windowInsets.getSystemWindowInsetRight()).thenReturn(100);
+    WindowInsets windowInsets =
+        new WindowInsets.Builder()
+            .setInsets(
+                android.view.WindowInsets.Type.navigationBars()
+                    | android.view.WindowInsets.Type.statusBars(),
+                Insets.of(100, 100, 100, 100))
+            .build();
     flutterView.onApplyWindowInsets(windowInsets);
 
     // Verify.
-    verify(flutterRenderer, times(2)).setViewportMetrics(viewportMetricsCaptor.capture());
-    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingTop);
-    // Padding bottom is always 0.
-    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingBottom);
+    verify(flutterRenderer, times(3)).setViewportMetrics(viewportMetricsCaptor.capture());
+    assertEquals(100, viewportMetricsCaptor.getValue().viewPaddingTop);
+    assertEquals(100, viewportMetricsCaptor.getValue().viewPaddingBottom);
     assertEquals(100, viewportMetricsCaptor.getValue().viewPaddingLeft);
     assertEquals(100, viewportMetricsCaptor.getValue().viewPaddingRight);
   }
 
   // This test uses the pre-API 30 Algorithm for window insets.
   @Test
-  @TargetApi(29)
+  @TargetApi(28)
```

Comment:
was there an error caused by target api = 29?

Reply:
The code behaves differently for API level 29 versus 28, and the test seems to be based on the API 28 implementation.